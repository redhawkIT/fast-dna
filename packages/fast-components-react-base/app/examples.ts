import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };
