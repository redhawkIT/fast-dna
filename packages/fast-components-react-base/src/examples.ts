import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}
