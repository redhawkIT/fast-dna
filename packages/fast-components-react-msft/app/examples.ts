import ButtonExamples from "../src/button/examples.data";
export {ButtonExamples};

import DividerExamples from "../src/divider/examples.data";
export {DividerExamples};
