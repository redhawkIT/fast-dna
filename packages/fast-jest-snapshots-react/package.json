{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "version": "1.6.0",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "private": true,
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "prepare": "tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "start": "webpack-dev-server",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.6.0",
    "@types/jest": "^22.2.2",
    "jest": "^22.4.3",
    "react": "^16.3.0",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "@types/react": "^16.3.0",
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
