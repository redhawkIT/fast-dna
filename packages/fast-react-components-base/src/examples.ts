import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: IExampleData<T>[];
}