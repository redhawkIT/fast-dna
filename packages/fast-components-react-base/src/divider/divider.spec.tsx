import * as React from "react";
import * as ShallowRenderer from "react-test-renderer/shallow";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import Divider, {
    DividerProps,
    DividerRoles,
    IDividerClassNameContract,
    IDividerHandledProps,
    IDividerManagedClasses,
    IDividerUnhandledProps
} from "./divider";

describe("divider snapshot", (): void => {
    generateSnapshots(examples);
});

configure({adapter: new Adapter()});

describe("divider", (): void => {
    const managedClasses: IDividerClassNameContract = {
        divider: "divider-class",
    };

    test("should have a displayName that matches the component name", () => {
        expect((Divider as any).name).toBe(Divider.displayName);
    });

    test("should not throw if managedClasses are not provided", () => {
        expect(
            () => {
                shallow(<Divider />);
            }
        ).not.toThrow();
    });

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: IDividerHandledProps & IDividerManagedClasses = {
            managedClasses
        };
        const unhandledProps: IDividerUnhandledProps = {
            "aria-hidden": true
        };
        const props: DividerProps = {...handledProps, ...unhandledProps};
        const rendered: any = shallow(
            <Divider {...props} />
        );

        expect(rendered.prop("aria-hidden")).not.toBe(undefined);
        expect(rendered.prop("aria-hidden")).toEqual(true);
    });

    test("should pass `role` prop when value is `presentation`", () => {
        const rendered: any = shallow(
            <Divider managedClasses={managedClasses} role={DividerRoles.presentation} />
        );

        expect(rendered.prop("role")).toEqual(DividerRoles.presentation);
    });

    test("should NOT have a role attribute when value is `separator`", () => {
        const rendered: any = shallow(
            <Divider managedClasses={managedClasses} role={DividerRoles.separator} />
        );

        expect(rendered.prop("role")).not.toEqual(DividerRoles.separator);
        expect(rendered.prop("role")).toBe(undefined);
    });
});
