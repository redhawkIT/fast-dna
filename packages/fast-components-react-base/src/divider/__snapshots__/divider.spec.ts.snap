// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`divider divider: 0 1`] = `
<hr
  className="divider"
/>
`;
